src/simp-algorithms/TRACE_Q.cpp

```diff
@@ -35,61 +35,90 @@ namespace trace_q {
 
                     auto range_queries = range_query_initialization(original_trajectory,
                                                                     x_coord, y_coord, t_interval, range_query_mbr);
-                    query_objects.insert(std::end(query_objects), std::begin(range_queries),
-                                         std::end(range_queries));
+                    query_objects.insert(std::end(query_objects), std::make_move_iterator(std::begin(range_queries)),
+                                         std::make_move_iterator(std::end(range_queries)));
                 }
             }
         }
 
-        auto knn_query_mbr = expand_MBR(calculate_MBR(original_trajectory), knn_query_grid_expansion_factor);
-
-        auto knn_query_points_on_axis = static_cast<int>(std::ceil(1 / knn_query_grid_density));
-        auto knn_query_time_points = static_cast<int>(std::ceil(1 / knn_query_time_interval_multiplier));
-
-        for (int x_point = 0; x_point <= knn_query_points_on_axis; ++x_point) {
-            // Scale the coordinate based on the grid density, the mbr and the current point on the x-axis.
-            auto x_coord = knn_query_mbr.x_low + x_point * knn_query_grid_density * (knn_query_mbr.x_high - knn_query_mbr.x_low);
-            for (int y_point = 0; y_point <= knn_query_points_on_axis; ++y_point) {
-                auto y_coord = knn_query_mbr.y_low + y_point * knn_query_grid_density * (knn_query_mbr.y_high - knn_query_mbr.y_low);
-
-                // Here we run the knn queries asynchronously for each query time point. Note that here, the query time points should not be larger than the allowed connections to the database (100)
-                std::vector<std::future<std::shared_ptr<spatial_queries::KNN_Query_Test>>> knn_futures{};
-                for (int t_point = 0; t_point <= knn_query_time_points; ++t_point) {
-                    auto t_interval = static_cast<long double>(knn_query_mbr.t_low) + t_point * knn_query_time_interval_multiplier
-                                                                                        * static_cast<long double>((knn_query_mbr.t_high - knn_query_mbr.t_low));
-
-                    knn_futures.emplace_back(std::async(std::launch::async, knn_query_initialization, x_coord, y_coord, t_interval, knn_query_mbr, knn_query_time_interval_multiplier, knn_k));
+        if (use_KNN_for_query_accuracy) {
+            auto knn_query_mbr = expand_MBR(calculate_MBR(original_trajectory), knn_query_grid_expansion_factor);
+
+            auto knn_query_points_on_axis = static_cast<int>(std::ceil(1 / knn_query_grid_density));
+            auto knn_query_time_points = static_cast<int>(std::ceil(1 / knn_query_time_interval_multiplier));
+
+            // Here we run the knn queries asynchronously for each query time point.
+            // Note that here, the query time points should not be larger than the allowed connections to the database (100)
```

Comment:
troede max var 50 og ikke 100?

Reply:
Det er max, men vi sætter en grænse på 50, da der kan åbnes connections af postgres' egen proces, som vi ikke kan styre. Så det er bare for at være på den sikre side og vi oplever stadig at programmet udnytter 100% af CPUen.

Reply:
Very cool